networkx/tests/test_relabel.py

```diff
@@ -147,10 +147,15 @@ def test_relabel_isolated_nodes_to_same(self):
         assert_nodes_equal(H.nodes(), list(range(4)))
 
     def test_relabel_nodes_missing(self):
-        with pytest.raises(KeyError):
-            G = nx.Graph([("A", "B"), ("A", "C"), ("B", "C"), ("C", "D")])
-            mapping = {0: "aardvark"}
-            G = nx.relabel_nodes(G, mapping, copy=False)
+        G = nx.Graph([("A", "B"), ("A", "C"), ("B", "C"), ("C", "D")])
+        mapping = {0: "aardvark"}
+        # copy=True
+        H = nx.relabel_nodes(G, mapping, copy=True)
+        assert_nodes_equal(H.nodes, G.nodes)
+        # copy=False
+        GG = G.copy()
+        nx.relabel_nodes(G, mapping, copy=False)
+        assert_nodes_equal(G.nodes, GG.nodes)
```

Comment:
This might be a nice one to parametrize, slap the `@pytest.mark.parametrize('copy', (True, False))` decorator on, adjust the signature: `def test_relabel_nodes_missing(self, copy)` then:

```suggestion
        G = nx.Graph([("A", "B"), ("A", "C"), ("B", "C"), ("C", "D")])
        mapping = {0: "aardvark"}
        H = nx.relabel_nodes(G, mapping, copy=copy)
        assert_nodes_equal(H.nodes, G.nodes)
```

Just a suggestion, feel free to ignore!

Reply:
Thanks!   I think the copy=False case need a different syntax due to G being changed in-place.  If we compare G to H we are actually comparing the same object.

Reply:
Great point :)